### DISCUSSION TITLE
关于 Statistic.Countdown 组件的一些疑问🤔

### DISCUSSION THREAD

**Discussion by Loving-Vincent** [ant-design/ant-design#28885] [Q&A]
背景：项目中的倒计时剩余10分钟时需要执行一个回调函数， `Statistic.Countdown` 组件没有提供这个API，索性就自己造轮子，昨天突然发现一个怪异的现象：当我从浏览器的其他tab页切换回来时，发现倒计时慢了好几分钟🤔。起初以为是Dan的 [useInterval hooks](https://overreacted.io/zh-hans/making-setinterval-declarative-with-react-hooks/) 的问题，最后发现是浏览器特性导致，这是我在  [现代JavaScript 教程](https://zh.javascript.info/settimeout-setinterval#zong-jie)  看到的解释
> 请注意，所有的调度方法都不能 保证 确切的延时。
> 
> 例如，浏览器内的计时器可能由于许多原因而变慢：
> 
> - CPU 过载。
> - 浏览器页签处于后台模式。
> - 笔记本电脑用的是电池供电（译注：使用电池供电会以降低性能为代价提升续航）。
> 
> 所有这些因素，可能会将定时器的最小计时器分辨率（最小延迟）增加到 300ms 甚至 1000ms，具体以浏览器及其设置为准。

但是我测试了antd的倒计时组件，切出页面在后台运行也不会有误差。把antd和我自己的组件一起使用，两个倒计时也都不会有误差，这一点让我很困惑。

1. 为什么 `Statistic.Countdown` 组件不受浏览器后台模式限制？
2. 为什么 `Statistic.Countdown` 组件定时器 [源码](https://github.com/ant-design/ant-design/blob/master/components/statistic/Countdown.tsx#L6) 中的执行间隔定义为 1000 / 30 ？

**Comment by Loving-Vincent** [comment]
> 为什么 Statistic.Countdown 组件定时器 源码 中的执行间隔定义为 1000 / 30 ？

这个应该是为了定时器组件展示毫秒的场景而设定的。

**Reply by yoyo837** [reply]
是不是刷新率太高没必要啊, 符合人眼能识别到的变动即可
